annessca/my-diary
# #159319112 Create new users and authenticate them

feat(user-authentication): authenticate endpoint with JSON web token 
- query database to find already existing data
 - add middleware to signup route
 - write tests to validate user signup

[Delivers #159319112]

I have been able to create a database using the command line. I created two relations, which are the users and the entries tables programmatically using Node.js. 

#### What does this PR do?
  * This pull request ensures that the signup endpoint is able to create new users
  * It ensures that users are valid
#### Description of Task to be completed?
  * Modify code to ensure endpoint is working
  * Test endpoint to make sure of how it is working
#### How should this be manually tested?
  * Repository should be cloned to the local machine
  * Create a database called diary-app and turn on the connection
  * Use postman to post data to the backend
#### Any background context you want to provide?
  * Throughout the building of the app, I have not had access to the GUI of the database
  * This means that I have not been able to view the content of the database table. I also always have issues with viewing records on the command line.
#### Questions: Is it safer to use Callbacks with PostgreSQL node than to use Promises?
                            How does the response body from the Postgres database display?
